src/class-wp-conditional-blocks.php

```diff
@@ -8,8 +8,137 @@
 namespace Alley\WP\WP_Conditional_Blocks;
 
 /**
- * Example Plugin
+ * WP_Conditional_Blocks
  */
 class WP_Conditional_Blocks {
+	/**
+	 * Singleton instance for this class.
+	 *
+	 * @var WP_Conditional_Blocks|null
+	 */
+	private static ?self $instance = null;
 
+	/**
+	 * Array of conditions.
+	 *
+	 * @var array
+	 */
+	private array $conditions = [];
+
+	/**
+	 * Private constructor.
+	 */
+	protected function __construct() {}
+
+	/**
+	 * Creates the singleton instance.
+	 *
+	 * @return self The singleton instance.
+	 */
+	public static function instance(): self {
```

Comment:
Do we want to use a singleton trait instead?

Reply:
Trait has been added.